ctfcli/cli/challenges.py

```diff
@@ -89,9 +91,10 @@ def restore(self, challenge=None):
             if url.endswith(".git"):
                 if challenge is not None and folder != challenge:
                     continue
-                click.echo(f"Cloning {url} to {folder}")
-                subprocess.call(["git", "clone", "--depth", "1", url, folder])
-                shutil.rmtree(str(Path(folder) / ".git"))
+                click.echo(f"Adding git repo {url} to {folder} as subtree")
+                git_remote = subprocess.check_output(["git", "remote", "show", url])
+                default_branch = re.findall('(?<=HEAD branch: )\w*', str(git_remote))[0]
```

Comment:
is there really no better way to do this? git subtree doesn't default to the default branch?

Reply:
Subtree expects the command to provide the remote branch reference. https://github.com/git/git/blob/master/contrib/subtree/git-subtree.txt#L13
If it's not provided it errors with: `You must provide <repository> <ref>`